pkg/kvstore/etcd.go

```diff
@@ -1034,15 +1039,10 @@ reList:
 			client.WithPrefix(), client.WithRev(nextRev))
 		for {
 			select {
-			case <-e.client.Ctx().Done():
-				return
-			case <-ctx.Done():
-				return
```

Comment:
> Additionally, I think that the same operations should also be performed in the select case above at line 968, otherwise the watcher is not cleaned up properly if the context is closed at that point.

yeah you are right.. 

> Do you think that it could be possible to preserve the handling of these two cases here, falling though to the stopWatch case?

yeah that would work as well.. (I am thinking `goto abort` in all these cases)

Reply:
I would suggest to just defer the cleanup logic, so that we are sure it is never missed (there is also another return case which is currently not handled).

Reply:
right, that's even better, thanks